src/geventhttpclient/connectionpool.py

```diff
@@ -103,7 +111,7 @@ def _create_socket(self):
 
             try:
                 sock.settimeout(self.connection_timeout)
-                sock.connect(sock_info[-1])
+                sock = self._connect_socket(sock, sock_info[-1])
```

Comment:
why is this change required ? Reverting back to the `sock.connect(sock_info[-1])` made our proxy `CONNECT` work. `CONNECT` was sent over `https` to squid proxy causing failure